packages/react-pages/src/node/index.ts

```diff
@@ -54,9 +55,25 @@ export default function pluginFactory(
         },
       },
     }),
-    configResolved(config) {
-      pagesDir = opts.pagesDir ?? path.resolve(config.root, 'pages')
+    configResolved({ root, plugins, logger }) {
+      pagesDir = opts.pagesDir ?? path.resolve(root, 'pages')
       pageStrategy = new PageStrategy(pagesDir, findPages, loadPageData)
+
+      // Inject parsing logic for frontmatter if missing.
+      const { devDependencies = {} } = require(path.join(root, 'package.json'))
+      if (!devDependencies['remark-frontmatter']) {
```

Comment:
It now checks the `devDependencies` instead of using `resolve-from` to check node_modules. This should fix false positives with flat node_modules. 👍 

I'm assuming `package.json` exists at the configured `root`.

Reply:
We should resolve `package.json` in the same way as vite:

https://github.com/vitejs/vite/blob/27785f7fcc5b45987b5f0bf308137ddbdd9f79ea/packages/vite/src/node/config.ts#L602

Reply:
We *could* neglect that until a user needs it.. :)